rita_client/src/exit_manager/exit_switcher.rs

```diff
@@ -454,7 +454,7 @@ fn get_exit_metrics(
         // All babel routes are advertised as /128, so we check if each 'single' ip is part of exit subnet
         let ip = route.prefix.ip();
 
-        if check_ip_in_subnet(ip, exits) && !blacklisted.contains(&ip) {
+        if check_if_ip_in_exit_list(ip, exit_list) && !blacklisted.contains(&ip) {
```

Comment:
do we still need the concept of blacklisting? I remember we used it mostly for rouge exits in the ip range, but it may also have use for just exits in the cluster that are not acting properly, which we do want this code to be robust against.

Reply:
Also can't we avoid iterating over routes and instead just iterate over the list of exits and lookup their router objects perhaps via a hashmap? This would be far more efficient, not that it probably matters in this case bu tit's good to optimize where possible.

Reply:
fixed